core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -361,11 +361,13 @@ static struct tee_mmap_region *init_xlation_table(struct tee_mmap_region *mm,
 	do  {
 		uint64_t desc = UNSET_DESC;
 
-		if ((mm->va + mm->size <= base_va) ||
-		    core_mmu_is_dynamic_vaspace(mm)) {
+		/* Skip unmmaped entries */
+		while (core_mmu_is_dynamic_vaspace(mm))
+			mm++;
```

Comment:
This should exit the `do {} while()` loop when reaching the end of the table (`mm->size==0)`.
Following is not really nice but should do the job:
```c
		while (mm->size && core_mmu_is_dynamic_vaspace(mm))
			mm++;
		if (!mm->size)
			break;
```